File: 6lec/F.cpp
#include <iostream>
#include <vector>
#include <math.h>
using namespace std;

bool Check(long long m, long long x, long long y, long long n) {
	m -= min(x, y);
	if (m < 0) return false;
	return (m / x + m / y + 1) >= n;
}

long long BinSearch2(long long l, long long r, long long x, long long y, long long n) {
	long long m;
	while (l < r) {
		m = (l + r) / 2;
		if (Check(m, x, y, n)) {
			r = m;
		}
		else {
			l = m + 1;
		}
	}
	return l;
}

int main()
{
	long long n, x, y;
	cin >> n >> x >> y;
	cout << BinSearch2(0, n*max(x,y), x, y, n);
}

File: 1lec/F.go
package main

import (
	"fmt"
	"math"
)

func minf(m map[int]int) int {
	min := 100000000
	mini := 0
	for i, v := range m {
		if min > i * v {
			min = i * v
			mini = i
		}
	}
	return mini
}

func main() {
	var (
		a, b, c, d int
	)
	fmt.Scan(&a, &b, &c, &d)
	m := make(map[int]int)
	m[a+c] = int(math.Max(float64(d), float64(b)))
	m[a+d] = int(math.Max(float64(b), float64(c)))
	m[b+c] = int(math.Max(float64(d), float64(a)))
	m[b+d] = int(math.Max(float64(a), float64(c)))
	p := minf(m)
	fmt.Print(p, m[p])
}